api/ai/new/interfaces/algorithm.py

```diff
@@ -63,15 +63,18 @@ def create_initial_teams(
                 ]
             )
 
+        largest_existing_id = max([t.id for t in initial_teams]) if initial_teams else 0
+        team_names = [t.name for t in initial_teams]
+
         # Create the remaining teams with unique names
-        counter = 1
+        new_team_id = largest_existing_id + 1
         while len(initial_teams) < team_generation_options.total_teams:
-            name = f"Team {counter}"
-            # todo: do this better, it loops through all names every time
-            if name not in map(lambda t: t.name, initial_teams):
-                # todo: unique id must be ensured now
-                initial_teams.append(Team(_id=-1, name=name))
-            counter += 1
+            name = f"Team {new_team_id}"
+            if name not in team_names:
```

Comment:
nit: This runs O(N), make the team_names to Set to makes it faster

Reply:
addressed - [70923df](https://github.com/Teamable-Analytics/algorithms/pull/173/commits/70923df96d0e7f004e8b8da4b4f8d09733f73ff1)